lib/spack/spack/concretize.py

```diff
@@ -190,10 +188,50 @@ def _concretize_task(packed_arguments: Tuple[int, str, TestsType]) -> Tuple[int,
     index, spec_str, tests = packed_arguments
     with tty.SuppressOutput(msg_enabled=False):
         start = time.time()
-        spec = Spec(spec_str).concretized(tests=tests)
+        spec = concretized(Spec(spec_str), tests=tests)
         return index, spec, time.time() - start
 
 
+def concretized(spec: Spec, tests: Union[bool, Iterable[str]] = False) -> Spec:
+    """Return a concretized copy of the given spec.
```

Comment:
I think we should think about the interface here. I would really just like to see *one* `concretize()` method in this package, with options for unification, etc.  If that delegates to the existing methods to start, that's fine, but right now I think there are too many methods in here and it's not clear how/when each one should be used.